file/src/main/java/akka/stream/alpakka/file/impl/DirectoryChangesSource.java

```diff
@@ -157,22 +158,26 @@ private void doPoll() {
               // overflow means that some file system change events may have been missed,
               // that may be ok for some scenarios but to make sure it does not pass unnoticed we
               // fail the stage
-              failStage(
-                  new RuntimeException("Overflow from watch service: '" + directoryPath + "'"));
-
+              //to prevent "overflow from watch service error" if there is an trigger on arriving  >~500 files
+              try {
+                Files.list(directoryPath)
+                    .filter(t -> t.toFile().isFile())
+                    .forEach(t -> buffer.add(combiner.apply(t, DirectoryChange.Creation))); //use "Creation" event by default
```

Comment:
So, on overflow, all files in the directory will be reported as `Creation` even files that were there before, right?

Reply:
Yes, you are right.